catalog/app/components/FileEditor/QuiltConfigEditor/QuiltSummarize/QuiltSummarize.tsx

```diff
@@ -0,0 +1,782 @@
+    (item: Listing.Item) => {
+      if (item.type === 'dir') {
+        setPath(item.to)
+      } else {
+        submit(item.to)
+      }
+    },
+    [submit],
+  )
+  return (
+    <>
+      <M.DialogContent>
+        <div className={classes.dialog}>
+          {data.case({
+            _: () => <FilePickerSkeleton />,
+            Ok: (res: requests.BucketListingResult) => (
+              <FilePicker initialPath={initialPath} res={res} onCell={handleCellClick} />
+            ),
+          })}
+        </div>
+      </M.DialogContent>
+      <M.DialogActions>
+        <M.Button onClick={onClose}>Cancel</M.Button>
+      </M.DialogActions>
+    </>
+  )
+}
+
+const useAddColumnStyles = M.makeStyles((t) => ({
+  root: {
+    animation: '$show 0.15s ease-out',
+    display: 'flex',
+    '&:last-child $divider': {
+      marginLeft: t.spacing(4),
+    },
+  },
+  inner: {
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'column',
+    position: 'relative',
+    '&:has($close:hover) $path': {
+      opacity: 0.3,
+    },
+    '&:has($close:hover) $extended': {
+      opacity: 0.3,
+    },
+  },
+  settings: {
+    margin: t.spacing(0, 1, -1, -0.5),
+    transition: 'transform 0.15s ease-out',
+    '&:hover': {
+      transform: 'rotate(180deg)',
+    },
+  },
+  extended: {
+    animation: '$slide 0.15s ease-out',
+    transition: 'opacity 0.3s ease-out',
+    paddingLeft: t.spacing(7),
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  expanded: {
+    background: t.palette.background.paper,
+    position: 'absolute',
+    right: '16px',
+    top: '4px',
+  },
+  path: {
+    transition: 'opacity 0.3s ease-out',
+    display: 'flex',
+    alignItems: 'flex-end',
+  },
+  field: {
+    marginTop: t.spacing(1),
+    minWidth: t.spacing(10),
+  },
+  divider: {
+    marginLeft: t.spacing(2),
+  },
+  render: {
+    border: `1px solid ${t.palette.divider}`,
+    borderRadius: t.shape.borderRadius,
+    marginTop: t.spacing(3),
+    padding: t.spacing(2),
+  },
+  select: {
+    marginTop: t.spacing(2),
+  },
+  toggle: {
+    padding: t.spacing(1, 0, 0),
+  },
+  close: {
+    position: 'absolute',
+    right: 0,
+    top: 0,
+  },
+  '@keyframes slide': {
+    from: {
+      opacity: 0,
+      transform: 'translateY(-8px)',
+    },
+    to: {
+      opacity: 1,
+      transform: 'translateY(0)',
+    },
+  },
+  '@keyframes show': {
+    from: {
+      opacity: 0,
+      transform: 'scale(0.9)',
+    },
+    to: {
+      opacity: 1,
+      transform: 'scale(1)',
+    },
+  },
+}))
+
+interface AddColumnProps {
+  className: string
+  column: Column
+  disabled?: boolean
+  onChange: React.Dispatch<React.SetStateAction<Layout>>
+  row: Row
+  last: boolean
+}
+
+function AddColumn({ className, column, disabled, last, onChange, row }: AddColumnProps) {
+  const { bucket, initialPath } = useParams()
+
+  const classes = useAddColumnStyles()
+  const { file } = column
+  // TODO: [simple, setSimple] mode instead of advanced
```

Comment:
When the user hides the rest of the fields in the form except `path`, I'm planning to clear them (asking confirmation first if they are not empty). Because, if users don't see input fields, they shouldn't expect to submit those values.
But if the user clicks to show these fields again, I'm going to fill these fields with previously entered fields.

So, instead of simple flag "show"/"hide", I wanted to use "show"/"hide, but there were these values here".

However, I don't know if I implement this anytime in the near future, so I'll remove this TODO note.

Reply:
i tend to use TODOs for short-term action items, which are to be resolved in the scope of the current PR, and XXX for longer term stuff or things requiring more consideration